test/util/client.go

```diff
@@ -63,65 +64,74 @@ func GetClusterAdminClientConfig(adminKubeConfigFile string) (*restclient.Config
 	return conf, nil
 }
 
-func GetClientForUser(clientConfig restclient.Config, username string) (*client.Client, *kclient.Client, *restclient.Config, error) {
-	token, err := tokencmd.RequestToken(&clientConfig, nil, username, "password")
+func GetClientForUser(adminClient client.Interface, clientConfig restclient.Config, username string) (client.Interface, kclient.Interface, *restclient.Config, error) {
+	user, err := getOrCreateUser(adminClient, username)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	userClientConfig := clientcmd.AnonymousClientConfig(&clientConfig)
-	userClientConfig.BearerToken = token
-
-	kubeClient, err := kclient.New(&userClientConfig)
-	if err != nil {
-		return nil, nil, nil, err
-	}
-
-	osClient, err := client.New(&userClientConfig)
+	token, err := getScopedTokenForUser(adminClient, user, []string{scope.UserFull})
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	return osClient, kubeClient, &userClientConfig, nil
+	return getClientForConfigAndToken(&clientConfig, token.Name)
 }
 
-func GetScopedClientForUser(adminClient *client.Client, clientConfig restclient.Config, username string, scopes []string) (*client.Client, *kclient.Client, *restclient.Config, error) {
-	// make sure the user exists
-	if _, _, _, err := GetClientForUser(clientConfig, username); err != nil {
-		return nil, nil, nil, err
+func getOrCreateUser(adminClient client.Interface, username string) (*userapi.User, error) {
+	user, err := adminClient.Users().Create(&userapi.User{ObjectMeta: kapi.ObjectMeta{Name: username}})
+	if err == nil {
+		return user, nil
 	}
-	user, err := adminClient.Users().Get(username)
-	if err != nil {
-		return nil, nil, nil, err
+	if kerrs.IsAlreadyExists(err) {
+		return adminClient.Users().Get(username)
 	}
+	return nil, err
+}
 
+func getScopedTokenForUser(adminClient client.Interface, user *userapi.User, scopes []string) (*oauthapi.OAuthAccessToken, error) {
 	token := &oauthapi.OAuthAccessToken{
-		ObjectMeta:  kapi.ObjectMeta{Name: fmt.Sprintf("%s-token-plus-some-padding-here-to-make-the-limit-%d", username, rand.Int())},
+		ObjectMeta:  kapi.ObjectMeta{Name: fmt.Sprintf("%s-token-plus-some-padding-here-to-make-the-limit-%d", user.Name, rand.Int())},
```

Comment:
auto-creating a token named like this is just one of the many reasons this should never be run on a production cluster